File: frontend/src/api/explain.ts
import { fetchClient } from "./client";
import type { ConvertResponse, ExplainResponse } from "../types/api";

export async function explainTranslation(
  translationData: ConvertResponse
): Promise<ExplainResponse> {
  return fetchClient<ExplainResponse>("/explain", {
    method: "POST",
    body: JSON.stringify({ translation_data: translationData }),
  });
}

File: backend/src/sprout_backend/api/routes/explain.py
from fastapi import APIRouter
from ...schemas.explain import ExplainRequest, ExplainResponse
from ...ai.client import generate_explanation
router = APIRouter()
@router.post("/explain", response_model=ExplainResponse)
async def explain_translation(request: ExplainRequest):
    """
    Takes the output from the /convert endpoint and returns
    a kid-friendly explanation of the Python code using OpenAI.
    """
    # Convert Pydantic model to dict for the AI client
    translation_dict = request.translation_data.model_dump()
    explanation = await generate_explanation(translation_dict)
    return explanation
